portfolio/src/main/webapp/script.js

```diff
@@ -83,9 +83,69 @@ for (i = 0; i < coll.length; i++) {
     });
 }
 
+var markers = [];
+
+var marker;
+
+// Marker information for notable restaurants
+var restaurants = [
+  ['Rich JC', 42.275201, -83.732619],
+  ['Frita Batidos', 42.280368, -83.749291],
+  ['Panda Express', 42.290879, -83.717648],
+  ['Piada', 42.278907, -83.740573],
+  ['Tomukun Korean BBQ', 42.279500, -83.742775]
+];
+
+var icons = {
+          michigan: {
+            icon: {
+                url: '/images/michigan_logo.png',
+                scaledSize: new google.maps.Size(32, 32),
+            }
+          }
+        };
+
 /** Creates a map and adds it to the page. */
-function createMap() {
+function initMap() {
```

Comment:
Not required here, but as a bonus refactor:

This seems like you're setting up a mix of floating globals and functions that operate on them. this seems like a candidate for object oriented programming (to encapsulate data and to avoid the global space pollution).

Consider the module pattern:
https://dzone.com/articles/module-pattern-in-javascript#:~:text=Module%20pattern%20emulates%20classes%20in,is%20the%20Revealing%20Module%20Pattern. is

(And bonus points for understanding why it works ;), though that's fairly advanced -- but incredibly useful! -- javascript)